samples/snippets/visualbasic/VS_Snippets_CLR/conceptual.disposable/vb/TryFinallyGenerated.vb

```diff
@@ -0,0 +1,19 @@
+﻿Imports System.IO
+
+Module TryFinallyGenerated
+    Public Sub Main()
+        Dim buffer(49) As Char
+        Dim streamReader As New StreamReader("File1.txt")
+        Try
+            Dim charsRead As Integer
+            Do While streamReader.Peek() <> -1
+                charsRead = streamReader.Read(buffer, 0, buffer.Length)
+                ' 
+                ' Process characters read.
+                '
+            Loop
+        Finally
+            If streamReader IsNot Nothing Then DirectCast(streamReader, IDisposable).Dispose()
```

Comment:
Does this not need an `End If`?

Reply:
Nope since it is a single line, I thought the same thing but just learn that too.